authors/apps/core/permissions.py

```diff
@@ -29,3 +29,19 @@ def has_object_permission(self, request, view, obj):
         # Rating permission is allowed to all other users except the owner
         # The article owner will only be allowed GET
         return obj.author != request.user
+
+
+class IsMyProfileOrReadOnly(BasePermission):
```

Comment:
The class IsAuthorOrReadOnly was created for this purpose, use that instead.

Reply:
Because my profile model is referencing the user model via a one-to-one relationship I had to override **obj.author != request.user**  in the IsAuthorOrReadOnly class with 
**obj.user.username == request.user.username**in the IsMyProfileOrReadOnly class